subgraphs/euler-finance/src/mappings/handlers.ts

```diff
@@ -1,170 +1,316 @@
+  const totalDepositBalance = bigIntChangeDecimals(poolSize.plus(totalBorrows), DEFAULT_DECIMALS, token.decimals);
+  if (totalBalances.gt(BIGINT_ZERO)) {
+    market.exchangeRate = bigIntToBDUseDecimals(totalDepositBalance, token.decimals).div(
+      bigIntToBDUseDecimals(totalBalances, DEFAULT_DECIMALS),
+    );
+  }
+
+  // tvl, totalDepositBalanceUSD and totalBorrowBalanceUSD may be updated again with new price
+  // in the loop below
+  const newTotalDepositBalanceUSD = bigIntToBDUseDecimals(totalDepositBalance, token.decimals).times(
+    token.lastPriceUSD!,
+  );
+  const newTotalBorrowBalanceUSD = bigIntToBDUseDecimals(totalBorrowBalance, token.decimals).times(token.lastPriceUSD!);
+  protocol.totalDepositBalanceUSD = protocol.totalDepositBalanceUSD
+    .plus(newTotalDepositBalanceUSD)
+    .minus(market.totalDepositBalanceUSD);
+  protocol.totalBorrowBalanceUSD = protocol.totalBorrowBalanceUSD
+    .plus(newTotalBorrowBalanceUSD)
+    .minus(market.totalBorrowBalanceUSD);
+  protocol.totalValueLockedUSD = protocol.totalDepositBalanceUSD;
+  protocol.save();
+
+  market.totalDepositBalanceUSD = newTotalDepositBalanceUSD;
+  market.totalBorrowBalanceUSD = newTotalBorrowBalanceUSD;
+  market.totalValueLockedUSD = market.totalDepositBalanceUSD;
+
+  market.inputTokenBalance = totalDepositBalance;
+  market.outputTokenSupply = totalBalances;
+  market._totalBorrowBalance = totalBorrowBalance;
+
+  const dTokenAddress = Address.fromString(market._dToken!);
+  const dToken = ERC20.bind(dTokenAddress);
+  // these should always equal
+  // totalBorrows == dTokenTotalSupply
+  if (totalBorrows.gt(BIGINT_ZERO)) {
+    const dTokenTotalSupply = dToken.totalSupply();
+    if (dTokenTotalSupply.gt(BIGINT_ZERO)) {
+      market._dTokenExchangeRate = bigIntToBDUseDecimals(totalBorrowBalance, token.decimals).div(
+        bigIntToBDUseDecimals(dTokenTotalSupply, DEFAULT_DECIMALS),
+      );
+    }
+  }
+  market.save();
+
+  if (interestRate.notEqual(assetStatus.interestRate)) {
+    // update interest rates if `interestRate` or `reserveFee` changed
+    updateInterestRates(market, interestRate, assetStatus.reserveFee, totalBorrows, totalBalances, event);
+  }
+  updateRevenue(reserveBalance, totalBalances, totalBorrows, protocol, market, assetStatus, event);
+  snapshotMarket(event.block, marketId, BIGDECIMAL_ZERO, null);
+
+  //for verification only
+  const eTokenAddress = Address.fromString(market.outputToken!);
+  const eToken = ERC20.bind(eTokenAddress);
+  const eTokenTotalSupply = eToken.totalSupply();
+  if (totalBalances.notEqual(eTokenTotalSupply)) {
+    log.warning("[logAssetStatus]market={}/{},totalBalances={},eTokenTotalSupply={}", [
+      market.name!,
+      market.id,
+      totalBalances.toString(),
+      eTokenTotalSupply.toString(),
+    ]);
+  }
+
+  // update prices, tvl, totalDepositBalanceUSD, totalBorrowBalanceUSD every 100 blocks (~20 min)
```

Comment:
can we make this between 10-15 mins. Our cron job that retrieves subgraph data from the subgraphs runs every 15 mins (ideally this would be updated every time)

Reply:
Changed to 75 blocks (~15 minutes)